doc/examples/transform/plot_masked_register_translation.py

```diff
@@ -0,0 +1,72 @@
+"""
+===================================
+Masked Normalized Cross-Correlation
+===================================
+
+In this example, we use the masked normalized cross-correlation to identify the relative shift
+between two similar images containing invalid data.
+
+In this case, the images cannot simply be masked before computing the cross-correlation, 
+as the masks will influence the computation. The influence of the masks must be removed from
+the cross-correlation, as is described in [1]_.
+
+In this example, we register the translation between two images. However, one of the 
+images has about 25% of the pixels which are corrupted.
+
+.. [1] D. Padfield, "Masked object registration in the Fourier domain" 
+       IEEE Transactions on Image Processing (2012). :DOI:`10.1109/TIP.2011.2181402`
+
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+
+from skimage import data
+from skimage.feature import masked_register_translation
+from scipy import ndimage as ndi
+
+image = data.camera()
+shift = (-22, 13)
+
+#############################################
+# Define areas of the image which are invalid.
+# Probability of an invalid pixel is 25%.
+# This could be due to a faulty detector, or edges that
+# are not affected by translation (e.g. moving object in a window).
+# See reference paper for more examples
+corrupted_pixels = np.random.choice([False, True], 
+                                    size = image.shape, 
+                                    p = [0.25, 0.75])
+
+# The shift corresponds to the pixel offset relative to the reference image
+offset_image = ndi.shift(image, shift)
+offset_image *= corrupted_pixels
+print("Known offset (y, x): {}".format(shift))
+
+# Determine what the mask is based on which pixels are invalid
+# In this case, we know what the mask should be since we corrupted 
+# the pixels ourselves
+mask = corrupted_pixels
+
+shift = masked_register_translation(image, offset_image, mask)
+
+print("Detected pixel offset (y, x): {}".format(shift))
```

Comment:
From the generated webpage http://scikit-image.org/docs/dev/auto_examples/transform/plot_masked_register_translation.html#sphx-glr-auto-examples-transform-plot-masked-register-translation-py:
```
Known offset (y, x): (-22, 13)
Detected pixel offset (y, x): [ 22. -13.]
```
Shouldn't they match?

Reply:
Following the logic of the `plot_register_translation.py` example ([link](http://scikit-image.org/docs/dev/auto_examples/transform/plot_register_translation.html#sphx-glr-auto-examples-transform-plot-register-translation-py)), the 'detected pixel offset' value is the shift required to _shift back_.

This makes sense when trying to correct for a shift, but for a gallery example, it is confusing. I've corrected this incongruence in #3528, so that they match.